packages/core/src/oidc/init.ts

```diff
@@ -198,6 +209,63 @@ export default function initOidc(envSet: EnvSet, queries: Queries, libraries: Li
       },
     },
     extraParams: [OIDCExtraParametersKey.InteractionMode],
+    extraTokenClaims: async (ctx, token) => {
+      try {
+        const { isDevFeaturesEnabled, isCloud } = EnvSet.values;
+        // No cloud connection for OSS version, skip.
+        if (!isDevFeaturesEnabled || !isCloud) {
+          return;
+        }
+
+        const isTokenClientCredentials = token instanceof ctx.oidc.provider.ClientCredentials;
+
+        const { script, envVars } =
+          (await trySafe(
+            logtoConfigs.getJwtCustomizer(
+              isTokenClientCredentials
+                ? LogtoJwtTokenKey.ClientCredentials
+                : LogtoJwtTokenKey.AccessToken
+            )
+          )) ?? {};
+
+        if (!script) {
+          return;
+        }
+
+        // Wait for cloud API to be ready and we can use cloud connection client to request the API.
+        const client = await cloudConnection.getClient();
+
+        // We pass context to the cloud API only when it is a user's access token.
+        const logtoUserInfo = conditional(
+          !isTokenClientCredentials &&
+            token.accountId &&
+            (await libraries.jwtCustomizers.getUserContext(token.accountId))
+        );
+        /**
+         * `token` and `context` can not be assigned to Record<string, Json> according to the type inference,
+         * use request body guard to ensure the type.
+         *
+         * Use direct type casting to avoid the type inference issue since if the type is not correct the client
+         * will throw an Zod type error, there is no need to implement the zod guard and error handling here.
+         */
+        // eslint-disable-next-line no-restricted-syntax
+        const payload = {
+          script,
+          envVars,
+          token,
+          context: conditional(logtoUserInfo && { user: logtoUserInfo }),
+        } as unknown as CustomJwtFetcher;
+        return (
+          (await trySafe(
+            client.post(`/api/services/custom-jwt`, {
+              body: payload,
+            })
```

Comment:
maybe need to design the error-handling logic further here. 

e.g. 
Define a new Error type for the external service call, like `CustomJwtRequestError. `

```js
const result = await trySafe(XXXX, (error: unknown) => {  if (xxx) { throw new CustomJwtRequestError()} else (throw error)  })
```

out side
```
try { ... }
catch(error) {
// do. some log here. 
}
```

Reply:
We already have log mechanism for cloud service so its OK for the current implementation.
<img width="862" alt="image" src="https://github.com/logto-io/logto/assets/15182327/5ad176c8-63c2-4d5c-81cc-85ed8a13a556">